hilti/toolchain/src/ast/builder/builder.cc

```diff
@@ -62,6 +62,18 @@ Expression* Builder::addTmp(const std::string& prefix, QualifiedType* t, Express
     return id(tmp);
 }
 
+Expression* Builder::addAttributeTmp(Attribute* attr, const std::string& prefix) {
```

Comment:
I'd say this is Spicy-specific functionality, so would not put it into the HILTI builder. Keeping it in the visitor would be one option if that's sufficient, otherwise the Spicy parser builder is be another common place for such helper functionality.